### PR TITLE
DEV-969 merge hathifiles-database, hathifiles-history, hathifiles

### PR DESCRIPTION
**Pull request by moseshll**
DEV-969 merge hathifiles-database, hathifiles-history, hathifiles
- Merge files from `hathifiles_history`
  - Adopt anonymous positional arguments (*) for Ruby 3.2
- Use `hathifile_database` gem.
- Remove `wait-for` and fetch using `wget` (will be superseded by healthcheck later)
- Remove `lib/hathifile_record.rb` which seems to do nothing except screw up tests when included.
- Rename `ci.yml` to `tests.yml`
- Remove local `bin/wait-for`

Currently `wait-for` doesn't seem to be findable by the GitHub action, that's why test is failing.

### CODE REVIEW

**Comment by aelkiss** [comment]
Yeah, let's get rid of `wait-for` and set up a proper health check as well (hopefully rinse & repeat from https://github.com/hathitrust/oai_solr/pull/27)

**Comment by moseshll** [comment]
Pushed the solution to the wait-for problem since I walked face-first into it. Have the healthcheck for mariadb working okay, pushgateway is going to take a little more work (it has a /healthy endpoint but still need to figure out how to invoke it).

File: README.md [hunk 1]
```diff
@@ -55,3 +55,162 @@ The following is a more precise definition of the MARC extractions.
 | 24 | Digitization Source | digitization_agent_code | The organization that digitized the content. 974$s |
 | 25 | Access profile | access_profile_code | Indicates whether an item has view or download restrictions. The `access_profile` field from the `rights_current` table for this htid. |
 | 26 | Author | author | 100$abcd, 110$abcd, **and** 111$acd stripped and uniqd, joined by ", " |
+
+## Catalog Redirect: Generate hathifile history and compute record redirects
+
+We want to generate redirects for catalog records that have been completely
+replaced by.
+
+## Developer Setup
```
Comment (aelkiss) [inline README.md:64]: Should be merged with general developer setup section.

**Review by aelkiss — COMMENTED** [review]
It looks like this is largely moving code from https://github.com/hathitrust/hathifile_history without much additional changes or code sharing. Based on the summaries in the commit messages this generally seems fine to me. I'm not seeing a lot of opportunities for code sharing from an initial look, but I am happy it can at least share the Docker setup etc. I did not attempt to run this locally (I'm happy with the tests running in github, and we can try this under argo workflows when ready).

Let me know if there are any specific areas or changes you'd like feedback on.

**Review by aelkiss — APPROVED** [review]
I think we can go ahead and merge this.

**Pull request merged**